src/Microsoft.OData.Edm/Csdl/CsdlJsonWriter.cs

```diff
@@ -84,9 +123,20 @@ EdmModelReferenceElementsJsonVisitor visitor
         }
 
         /// <summary>
-        /// Schema Object.
+        /// Asynchronously writes $Reference Object.
         /// </summary>
-        private void WriteSchemata()
+        /// <returns>A task that represents the asynchronous operation</returns>
+        private Task WriteReferenceElementsAsync()

```

Comment:
no async key word for this private method?

Reply:
@xuzhg This is the method that is in question to either rename it to `WriteSchemas` or maintain it as `WriteSchemata`. The corresponding async method is here this 
[async Task WriteSchemasAsync()](https://github.com/OData/odata.net/blob/fix/2684-csdlwriter-async-apis/src/Microsoft.OData.Edm/Csdl/CsdlJsonWriter.cs#L157)